File: linked-list/delete-nodes-from-linked-list-present-in-array.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode modifiedList(int[] nums, ListNode head) {
        HashMap<Integer, Boolean> map = new HashMap<>();
        for(int element: nums){
            if(map.get(element) == null){
                map.put(element,true);
            }
        }
        ListNode dummy = new ListNode(0);
        ListNode ans = dummy;
        while(head != null){
            if(map.get(head.val) == null){
                ListNode node = new ListNode(head.val);
                dummy.next = node;
                dummy = dummy.next;
            }
            head = head.next;
        }
        return ans.next;
    }
}

File: linked-list/delete-nodes-from-linked-list-present-in-array.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def modifiedList(self, nums: List[int], head: Optional[ListNode]) -> Optional[ListNode]:
        d = dict()
        for ele in nums:
            if d.get(ele) == None:
                d[ele] = 1
        

        temp = ListNode()
        ans = temp
        while head != None:
            if d.get(head.val) == None:
                node = ListNode(head.val)
                temp.next = node
                temp = temp.next
            head = head.next
        return ans.next